core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java

```diff
@@ -79,15 +80,39 @@ public synchronized  InodeDirectory getParentInodeDirectory()
     if (mDescendantInode == null) {
       return super.getParentInodeDirectory();
     }
-    throw new UnsupportedOperationException();
+    try {
+      if (super.getInode().getId() == mDescendantInode.getParentId()) {
+        Inode<?> parentInode = super.getInode();
```

Comment:
NIT: Add a comment that the descendant is direct child, so getting parent is possible.